src/visitor/RewriteInputPropVisitor.cpp

```diff
@@ -121,42 +121,41 @@ void RewriteInputPropVisitor::visit(VariablePropertyExpression* expr) {
 }
 
 void RewriteInputPropVisitor::visit(ListExpression* expr) {
-    auto items = std::move(*expr).get();
-    for (auto iter = items.begin(); iter < items.end(); ++iter) {
-        iter->get()->accept(this);
+    const auto& items = expr->items();
+    for (size_t i = 0; i < items.size(); ++i) {
+        const_cast<Expression*>(items[i])->accept(this);
         if (ok()) {
-            *iter = std::move(result_);
+            expr->setItem(i, result_.release());
         }
     }
-    expr->setItems(std::move(items));
 }
 
 void RewriteInputPropVisitor::visit(SetExpression* expr) {
-    auto items = std::move(*expr).get();
-    for (auto iter = items.begin(); iter < items.end(); ++iter) {
-        iter->get()->accept(this);
+    const auto& items = expr->items();
+    for (size_t i = 0; i < items.size(); ++i) {
+        const_cast<Expression*>(items[i])->accept(this);
         if (ok()) {
-            *iter = std::move(result_);
+            expr->setItem(i, result_.release());
```

Comment:
Not sure if it would be helpful if we add such an interface
```
setItem(size_t index, std::unique_ptr<Expression> item)
```

Reply:
Good suggestion. Fixed in vesoft-inc/nebula-common#268